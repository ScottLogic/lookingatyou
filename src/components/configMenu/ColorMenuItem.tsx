/* tslint:disable: jsx-no-lambda */

import React from 'react';
export default function TextBoxMenuItem(props: {
    name: string;
    onInputChange: (text: string) => void;
    color: string;
}) {
    return (
        <div>
            <label>{props.name}</label>
            <input
                type="color"
                value={props.color}
                onChange={event => {
                    props.onInputChange(event.target.value);
                }}
            />
        </div>
    );
}
