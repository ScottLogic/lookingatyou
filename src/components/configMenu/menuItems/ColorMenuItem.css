.colorPicker-content {
    width: fit-content !important;
}

.colorPicker-overlay {
    background: rgba(0, 0, 0, 0.2) !important;
}

.launchColorPicker {
    width: 70%;
    position: absolute;
    left: 15%;
}

.launchColorPicker > span {
    background: inherit;
    background-clip: text;
    color: transparent;
    filter: invert(1) grayscale(1) contrast(9);
}

.closeColorPicker {
    position: absolute;
    width: 50%;
    left: 25%;
}
