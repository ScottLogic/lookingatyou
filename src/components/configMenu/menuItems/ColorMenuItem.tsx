import React from 'react';
import { HelpWith } from '../Help';

export interface IColorMenuItemProps {
    name: string;
    onInputChange: (text: string) => void;
    color: string;
    helpWith: HelpWith;
}

const ColorMenuItem = React.memo(
    (props: IColorMenuItemProps) => {
        function onChange(event: React.ChangeEvent<HTMLInputElement>) {
            props.onInputChange(event.target.value);
        }
        return (
            <div data-tip={true} data-for={HelpWith[props.helpWith]}>
                <label>{props.name}</label>
                <input type="color" value={props.color} onChange={onChange} />
            </div>
        );
    },
    (previous, next) =>
        previous.color === next.color && previous.name === next.name,
);
export default ColorMenuItem;
