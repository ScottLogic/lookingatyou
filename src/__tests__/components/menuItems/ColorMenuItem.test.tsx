import { mount, shallow } from 'enzyme';
import React from 'react';
import ColorMenuItem, {
    IColorMenuItemProps,
} from '../../../components/configMenu/menuItems/ColorMenuItem';

let props: IColorMenuItemProps;
let mockOnInputChange: jest.Mock;

describe('TextBoxMenuItem tests', () => {
    beforeEach(() => {
        mockOnInputChange = jest.fn();
        props = {
            name: 'test',
            color: '#cbcbcb',
            onInputChange: mockOnInputChange,
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(<ColorMenuItem {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onInputChange on change event', () => {
        const wrapper = mount(<ColorMenuItem {...props} />);
        wrapper
            .find('input')
            .simulate('change', { target: { value: '#ABABAB' } });
        expect(mockOnInputChange).toBeCalledTimes(1);
    });
});
