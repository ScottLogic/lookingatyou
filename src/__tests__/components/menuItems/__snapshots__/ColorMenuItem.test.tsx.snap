// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBoxMenuItem tests should render correctly 1`] = `
"<div data-tip={true} data-for={[undefined]}>
  <label>
    test
  </label>
  <input type=\\"color\\" value=\\"#cbcbcb\\" onChange={[Function: onChange]} />
</div>"
`;
